Index: pkg-php/TSRM/configure.in
===================================================================
--- pkg-php.orig/TSRM/configure.in	2009-08-25 15:43:42.000000000 +0200
+++ pkg-php/TSRM/configure.in	2009-08-25 15:44:02.000000000 +0200
@@ -13,9 +13,6 @@
 TSRM_THREADS_CHECKS
 	
 AM_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  AM_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl TSRM_PTHREAD
 
Index: pkg-php/configure.in
===================================================================
--- pkg-php.orig/configure.in	2009-08-25 15:43:42.000000000 +0200
+++ pkg-php/configure.in	2009-08-25 15:44:43.000000000 +0200
@@ -1357,9 +1357,6 @@
 ])
 AC_PROG_LIBTOOL
 
-if test "$enable_debug" != "yes"; then
-  PHP_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl libtool 1.4.3 needs this.
 PHP_SET_LIBTOOL_VARIABLE([--preserve-dup-deps])
