Index: php5-5.2.3/TSRM/configure.in
===================================================================
--- php5-5.2.3.orig/TSRM/configure.in	2007-06-02 13:59:06.000000000 +0200
+++ php5-5.2.3/TSRM/configure.in	2007-06-02 13:59:09.000000000 +0200
@@ -13,9 +13,6 @@
 TSRM_THREADS_CHECKS
 	
 AM_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  AM_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl TSRM_PTHREAD
 
Index: php5-5.2.3/configure.in
===================================================================
--- php5-5.2.3.orig/configure.in	2007-06-02 13:59:06.000000000 +0200
+++ php5-5.2.3/configure.in	2007-06-02 13:59:09.000000000 +0200
@@ -1273,9 +1273,6 @@
 
 LDFLAGS="$LDFLAGS $PHP_AIX_LDFLAGS"
 AC_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  PHP_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl libtool 1.4.3 needs this.
 PHP_SET_LIBTOOL_VARIABLE([--preserve-dup-deps])
