Index: php5-5.2.4/TSRM/configure.in
===================================================================
--- php5-5.2.4.orig/TSRM/configure.in	2007-09-10 20:45:24.000000000 +0200
+++ php5-5.2.4/TSRM/configure.in	2007-09-10 20:45:26.000000000 +0200
@@ -13,9 +13,6 @@
 TSRM_THREADS_CHECKS
 	
 AM_PROG_LIBTOOL
-if test "$enable_debug" != "yes"; then
-  AM_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl TSRM_PTHREAD
 
Index: php5-5.2.4/configure.in
===================================================================
--- php5-5.2.4.orig/configure.in	2007-09-10 20:45:24.000000000 +0200
+++ php5-5.2.4/configure.in	2007-09-10 20:45:26.000000000 +0200
@@ -1256,9 +1256,6 @@
 AC_DEFUN([AC_PROG_CXX], [])])
 AC_PROG_LIBTOOL
 
-if test "$enable_debug" != "yes"; then
-  PHP_SET_LIBTOOL_VARIABLE([--silent])
-fi
 
 dnl libtool 1.4.3 needs this.
 PHP_SET_LIBTOOL_VARIABLE([--preserve-dup-deps])
