From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Sun, 1 Mar 2015 10:41:50 +0100
Subject: 047-zts_with_dl

---
 ext/standard/dl.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/ext/standard/dl.c b/ext/standard/dl.c
index 9bf3e44..c441d81 100644
--- a/ext/standard/dl.c
+++ b/ext/standard/dl.c
@@ -74,12 +74,7 @@ PHPAPI PHP_FUNCTION(dl)
 		(strcmp(sapi_module.name, "cli") != 0) &&
 		(strncmp(sapi_module.name, "embed", 5) != 0)
 	) {
-#ifdef ZTS
-		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Not supported in multithreaded Web servers - use extension=%s in your php.ini", filename);
-		RETURN_FALSE;
-#else
 		php_error_docref(NULL TSRMLS_CC, E_DEPRECATED, "dl() is deprecated - use extension=%s in your php.ini", filename);
-#endif
 	}
 
 	php_dl(filename, MODULE_TEMPORARY, return_value, 0 TSRMLS_CC);
