Index: php5-5.2.4/acinclude.m4
===================================================================
--- php5-5.2.4.orig/acinclude.m4	2007-09-10 20:45:23.000000000 +0200
+++ php5-5.2.4/acinclude.m4	2007-09-10 20:45:32.000000000 +0200
@@ -2364,9 +2364,7 @@
 
     PHP_ADD_INCLUDE($OPENSSL_INCDIR)
   
-    PHP_CHECK_LIBRARY(crypto, CRYPTO_free, [
-      PHP_ADD_LIBRARY(crypto,,$1)
-    ],[
+    PHP_CHECK_LIBRARY(crypto, CRYPTO_free, [:],[
       AC_MSG_ERROR([libcrypto not found!])
     ],[
       -L$OPENSSL_LIBDIR
