Index: php5-5.2.2/ext/ldap/ldap.c
===================================================================
--- php5-5.2.2.orig/ext/ldap/ldap.c	2007-05-04 17:08:08.000000000 +0200
+++ php5-5.2.2/ext/ldap/ldap.c	2007-05-04 17:08:23.000000000 +0200
@@ -1320,7 +1320,7 @@
 	}
 
 	i=0;
-	while (ldap_value[i] != NULL) i++;
+	while (ldap_value && ldap_value[i] != NULL) i++;
 	count = i;
 
 	array_init(return_value);
@@ -1330,7 +1330,8 @@
 		add_index_string(return_value, i, ldap_value[i], 1);
 	}
 
-	ldap_value_free(ldap_value);
+	if (ldap_value)
+		ldap_value_free(ldap_value);
 }
 /* }}} */
 
