Index: php5-5.2.4/ext/ldap/ldap.c
===================================================================
--- php5-5.2.4.orig/ext/ldap/ldap.c	2007-09-10 20:45:23.000000000 +0200
+++ php5-5.2.4/ext/ldap/ldap.c	2007-09-10 20:45:39.000000000 +0200
@@ -1334,7 +1334,7 @@
 	}
 
 	i=0;
-	while (ldap_value[i] != NULL) i++;
+	while (ldap_value && ldap_value[i] != NULL) i++;
 	count = i;
 
 	array_init(return_value);
@@ -1344,7 +1344,8 @@
 		add_index_string(return_value, i, ldap_value[i], 1);
 	}
 
-	ldap_value_free(ldap_value);
+	if (ldap_value)
+		ldap_value_free(ldap_value);
 }
 /* }}} */
 
