Index: php5-5.2.6/ext/ldap/ldap.c
===================================================================
--- php5-5.2.6.orig/ext/ldap/ldap.c
+++ php5-5.2.6/ext/ldap/ldap.c
@@ -1334,7 +1334,7 @@ PHP_FUNCTION(ldap_explode_dn)
 	}
 
 	i=0;
-	while (ldap_value[i] != NULL) i++;
+	while (ldap_value && ldap_value[i] != NULL) i++;
 	count = i;
 
 	array_init(return_value);
@@ -1344,7 +1344,8 @@ PHP_FUNCTION(ldap_explode_dn)
 		add_index_string(return_value, i, ldap_value[i], 1);
 	}
 
-	ldap_value_free(ldap_value);
+	if (ldap_value)
+		ldap_value_free(ldap_value);
 }
 /* }}} */
 
