Index: pkg-php/ext/ldap/ldap.c
===================================================================
--- pkg-php.orig/ext/ldap/ldap.c	2009-08-25 15:43:42.000000000 +0200
+++ pkg-php/ext/ldap/ldap.c	2009-08-25 15:44:02.000000000 +0200
@@ -1239,7 +1239,7 @@
 	}
 
 	i=0;
-	while (ldap_value[i] != NULL) i++;
+	while (ldap_value && ldap_value[i] != NULL) i++;
 	count = i;
 
 	array_init(return_value);
@@ -1249,7 +1249,8 @@
 		add_index_string(return_value, i, ldap_value[i], 1);
 	}
 
-	ldap_value_free(ldap_value);
+	if (ldap_value)
+		ldap_value_free(ldap_value);
 }
 /* }}} */
 
