From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Sun, 1 Mar 2015 10:41:54 +0100
Subject: hurd-noptrace

---
 sapi/fpm/config.m4 | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sapi/fpm/config.m4 b/sapi/fpm/config.m4
index 579f7c2..9dd09d8 100644
--- a/sapi/fpm/config.m4
+++ b/sapi/fpm/config.m4
@@ -147,6 +147,11 @@ AC_DEFUN([AC_FPM_TRACE],
         pid_t child;
         int status;
 
+        /* broken ptrace on Hurd, avoid hanging */
+        #ifdef __GNU__
+        return 10;
+        #endif
+
         if ( (child = fork()) ) { /* parent */
           int ret = 0;
 
