From: Xinchen Hui <laruence@php.net>
Date: Thu, 8 Jan 2015 16:09:02 +0800
Subject: Fixed bug #68741 (Null pointer deference in ext/pgsql/pgsql.c)
 (CVE-2015-1352)

---
 ext/pgsql/pgsql.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/pgsql/pgsql.c b/ext/pgsql/pgsql.c
index 2e73da0..62bc6d3 100644
--- a/ext/pgsql/pgsql.c
+++ b/ext/pgsql/pgsql.c
@@ -6117,6 +6117,9 @@ static inline void build_tablename(smart_str *querystr, PGconn *pg_link, const c
 	/* schame.table should be "schame"."table" */
 	table_copy = estrdup(table);
 	token = php_strtok_r(table_copy, ".", &tmp);
+	if (token == NULL) {
+		token = table;
+	}
 	len = strlen(token);
 	if (_php_pgsql_detect_identifier_escape(token, len) == SUCCESS) {
 		smart_str_appendl(querystr, token, len);
