Index: php5-5.2.4/acinclude.m4
===================================================================
--- php5-5.2.4.orig/acinclude.m4	2007-09-10 20:45:32.000000000 +0200
+++ php5-5.2.4/acinclude.m4	2007-09-11 00:24:05.000000000 +0200
@@ -1210,7 +1210,7 @@
     }
 
   ],[
-    ac_cv_pwrite=yes
+    ac_cv_pwrite=no
   ],[
     ac_cv_pwrite=no
   ],[
@@ -1239,7 +1239,7 @@
     exit(0);
     }
   ],[
-    ac_cv_pread=yes
+    ac_cv_pread=no
   ],[
     ac_cv_pread=no
   ],[
