Index: pkg-php/acinclude.m4
===================================================================
--- pkg-php.orig/acinclude.m4	2009-08-25 15:44:02.000000000 +0200
+++ pkg-php/acinclude.m4	2009-08-25 15:44:02.000000000 +0200
@@ -1219,7 +1219,7 @@
     }
 
   ],[
-    ac_cv_pwrite=yes
+    ac_cv_pwrite=no
   ],[
     ac_cv_pwrite=no
   ],[
@@ -1248,7 +1248,7 @@
     exit(0);
     }
   ],[
-    ac_cv_pread=yes
+    ac_cv_pread=no
   ],[
     ac_cv_pread=no
   ],[
