Index: php5-5.2.0/acinclude.m4
===================================================================
--- php5-5.2.0.orig/acinclude.m4	2007-03-18 22:58:39.000000000 +0100
+++ php5-5.2.0/acinclude.m4	2007-03-18 22:58:40.000000000 +0100
@@ -1134,7 +1134,7 @@
     }
 
   ],[
-    ac_cv_pwrite=yes
+    ac_cv_pwrite=no
   ],[
     ac_cv_pwrite=no
   ],[
@@ -1163,7 +1163,7 @@
     exit(0);
     }
   ],[
-    ac_cv_pread=yes
+    ac_cv_pread=no
   ],[
     ac_cv_pread=no
   ],[
