Index: php5-5.2.6/acinclude.m4
===================================================================
--- php5-5.2.6.orig/acinclude.m4
+++ php5-5.2.6/acinclude.m4
@@ -1210,7 +1210,7 @@ $1
     }
 
   ],[
-    ac_cv_pwrite=yes
+    ac_cv_pwrite=no
   ],[
     ac_cv_pwrite=no
   ],[
@@ -1239,7 +1239,7 @@ $1
     exit(0);
     }
   ],[
-    ac_cv_pread=yes
+    ac_cv_pread=no
   ],[
     ac_cv_pread=no
   ],[
