--- php4-4.3.9/ext/readline/config.m4	2003-09-30 20:54:04.000000000 -0600
+++ php4-4.3.9/ext/readline/config.m4	2004-10-04 21:40:38.000000000 -0600
@@ -12,7 +12,7 @@
 
 if test "$PHP_READLINE" && test "$PHP_READLINE" != "no"; then
   for i in $PHP_READLINE /usr/local /usr; do
-    test -f $i/include/readline/readline.h && READLINE_DIR=$i && break
+    test -f $i/include/editline/readline.h && READLINE_DIR=$i && break
   done
 
   if test -z "$READLINE_DIR"; then
@@ -64,7 +64,7 @@
 elif test "$PHP_LIBEDIT" != "no"; then
 
   for i in $PHP_LIBEDIT /usr/local /usr; do
-    test -f $i/include/readline/readline.h && LIBEDIT_DIR=$i && break
+    test -f $i/include/editline/readline.h && LIBEDIT_DIR=$i && break
   done
 
   if test -z "$LIBEDIT_DIR"; then
--- php4-4.3.9/ext/readline/readline.c	2002-12-31 09:35:15.000000000 -0700
+++ php4-4.3.9/ext/readline/readline.c	2004-10-04 21:42:15.000000000 -0600
@@ -33,9 +33,9 @@
 #define rl_completion_matches completion_matches
 #endif
 
-#include <readline/readline.h>
+#include <editline/readline.h>
 #ifndef HAVE_LIBEDIT
-#include <readline/history.h>
+#include <editline/history.h>
 #endif
 
 PHP_FUNCTION(readline);
