Index: pkg-php/ext/sockets/sockets.c
===================================================================
--- pkg-php.orig/ext/sockets/sockets.c	2009-08-25 15:43:41.000000000 +0200
+++ pkg-php/ext/sockets/sockets.c	2009-08-25 15:44:03.000000000 +0200
@@ -708,6 +708,7 @@
 
 		php_sock = (php_socket*) zend_fetch_resource(element TSRMLS_CC, -1, le_socket_name, NULL, 1, le_socket);
 		if (!php_sock) continue; /* If element is not a resource, skip it */
+		if (php_sock->bsd_socket > FD_SETSIZE) continue; /* must ignore it */
 
 		PHP_SAFE_FD_SET(php_sock->bsd_socket, fds);
 		if (php_sock->bsd_socket > *max_fd) {
Index: pkg-php/ext/standard/streamsfuncs.c
===================================================================
--- pkg-php.orig/ext/standard/streamsfuncs.c	2009-08-25 15:43:41.000000000 +0200
+++ pkg-php/ext/standard/streamsfuncs.c	2009-08-25 15:44:03.000000000 +0200
@@ -612,6 +612,9 @@
 		 * is not displayed.
 		 * */
 		if (SUCCESS == php_stream_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL, (void*)&this_fd, 1) && this_fd >= 0) {
+			if (this_fd > FD_SETSIZE)
+				continue;
+
 			
 			PHP_SAFE_FD_SET(this_fd, fds);
 
