Index: php5-5.2.4/ext/sockets/sockets.c
===================================================================
--- php5-5.2.4.orig/ext/sockets/sockets.c	2007-07-24 13:35:08.000000000 +0200
+++ php5-5.2.4/ext/sockets/sockets.c	2007-09-11 00:24:23.000000000 +0200
@@ -566,6 +566,7 @@
 
 		php_sock = (php_socket*) zend_fetch_resource(element TSRMLS_CC, -1, le_socket_name, NULL, 1, le_socket);
 		if (!php_sock) continue; /* If element is not a resource, skip it */
+		if (php_sock->bsd_socket > FD_SETSIZE) continue; /* must ignore it */
 
 		PHP_SAFE_FD_SET(php_sock->bsd_socket, fds);
 		if (php_sock->bsd_socket > *max_fd) {
Index: php5-5.2.4/ext/standard/streamsfuncs.c
===================================================================
--- php5-5.2.4.orig/ext/standard/streamsfuncs.c	2007-07-09 19:27:24.000000000 +0200
+++ php5-5.2.4/ext/standard/streamsfuncs.c	2007-09-11 00:24:23.000000000 +0200
@@ -592,6 +592,9 @@
 		 * is not displayed.
 		 * */
 		if (SUCCESS == php_stream_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL, (void*)&this_fd, 1) && this_fd >= 0) {
+			if (this_fd > FD_SETSIZE)
+				continue;
+
 			
 			PHP_SAFE_FD_SET(this_fd, fds);
 
