Index: php5-5.2.4/ext/recode/recode.c
===================================================================
--- php5-5.2.4.orig/ext/recode/recode.c	2007-06-22 02:02:15.000000000 +0200
+++ php5-5.2.4/ext/recode/recode.c	2007-09-11 00:26:24.000000000 +0200
@@ -133,7 +133,7 @@
 	RECODE_REQUEST request = NULL;
 	char *r = NULL;
 	size_t r_len = 0, r_alen = 0;
-	int req_len, str_len;
+	size_t req_len, str_len;
 	char *req, *str;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE) {
