Description: Check for possible overflows in recode_string()
Author: Sean Finney <seanius@debian.org>
Bug-Debian: http://bugs.debian.org/459020
Bug: https://bugs.php.net/bug.php?id=41765
Forwarded: no
Last-Update: 2010-01-18

--- php5.orig/ext/recode/recode.c
+++ php5/ext/recode/recode.c
@@ -149,7 +149,7 @@ PHP_FUNCTION(recode_string)
 	int req_len, str_len;
 	char *req, *str;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE || str_len < 0) {
 		return;
 	}
 
