--- php-5.0.4/ext/recode/recode.c	2005-07-13 14:43:06.000000000 +1000
+++ php-5.0.4/ext/recode/recode.c	2005-07-13 14:43:45.000000000 +1000
@@ -129,8 +129,8 @@
 {
 	RECODE_REQUEST request = NULL;
 	char *r = NULL;
-	int r_len = 0, r_alen = 0;
-	int req_len, str_len;
+	size_t r_len = 0, r_alen = 0;
+	size_t req_len, str_len;
 	char *req, *str;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE) {
