--- php5-5.1.1/ext/standard/dl.c	2005-08-09 02:49:43.000000000 +1000
+++ php5-5.1.1/ext/standard/dl.c	2005-12-15 15:40:28.000000000 +1100
@@ -76,12 +76,7 @@
 	if ((strncmp(sapi_module.name, "cgi", 3)!=0) && 
 		(strcmp(sapi_module.name, "cli")!=0) &&
 		(strncmp(sapi_module.name, "embed", 5)!=0)) {
-#ifdef ZTS
-		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Not supported in multithreaded Web servers - use extension=%s in your php.ini", Z_STRVAL_PP(file));
-		RETURN_FALSE;
-#else
 		php_error_docref(NULL TSRMLS_CC, E_STRICT, "dl() is deprecated - use extension=%s in your php.ini", Z_STRVAL_PP(file));
-#endif
 	}
 
 	php_dl(*file, MODULE_TEMPORARY, return_value, 0 TSRMLS_CC);
