diff -uNr php-5.0.4.orig/scripts/Makefile.frag php-5.0.4/scripts/Makefile.frag
--- php-5.0.4.orig/scripts/Makefile.frag	2005-02-28 06:28:25.000000000 +0100
+++ php-5.0.4/scripts/Makefile.frag	2005-04-26 11:48:18.421951904 +0200
@@ -38,17 +38,17 @@
 	main/ \
 	main/streams/ \
 	regex/ \
+	ext/gd \
 	ext/iconv/ \
 	ext/libxml/ \
 	ext/standard/ \
 	ext/session/ \
 	ext/dom/xml_common.h \
 	ext/xml/ \
-	ext/xml/expat/ \
 	ext/mbstring/ \
 	ext/mbstring/libmbfl/ \
-    ext/mbstring/libmbfl/mbfl/ \
+	ext/mbstring/libmbfl/mbfl/ \
+	ext/mbstring/mbregex \
-	ext/sqlite/libsqlite/src/sqlite.h
 
 install-headers:
 	-@for i in $(HEADER_DIRS); do \
