Index: php5-5.2.2/configure.in
===================================================================
--- php5-5.2.2.orig/configure.in	2007-05-04 17:08:32.000000000 +0200
+++ php5-5.2.2/configure.in	2007-05-04 17:42:39.000000000 +0200
@@ -1082,8 +1082,13 @@
 
 ZEND_MODULE_API_NO=`$EGREP '#define ZEND_MODULE_API_NO ' $srcdir/Zend/zend_modules.h|$SED 's/#define ZEND_MODULE_API_NO //'`
 
+DEBIAN_PHP_API=`egrep -h '^#define ZEND_EXTENSION_API_NO|^#define ZEND_MODULE_API_NO|#define PHP_API_VERSION' $srcdir/Zend/zend_extensions.h $srcdir/Zend/zend_modules.h $srcdir/main/php.h | awk '{print $3}' | sed -e 's/^2200/200/' | sort -n | tail -n 1`
+if echo "$CPPFLAGS $CFLAGS" | grep -q -- -D_FILE_OFFSET_BITS=64; then
+  DEBIAN_PHP_API="${DEBIAN_PHP_API}+lfs"
+fi
+
 if test -z "$EXTENSION_DIR"; then
-  extbasedir=$ZEND_MODULE_API_NO
+  extbasedir=$DEBIAN_PHP_API
   if test "$oldstyleextdir" = "yes"; then
     if test "$PHP_DEBUG" = "1"; then
       part1=debug
@@ -1222,6 +1227,7 @@
 PHP_SUBST(CXXFLAGS)
 PHP_SUBST(CXXFLAGS_CLEAN)
 PHP_SUBST_OLD(DEBUG_CFLAGS)
+PHP_SUBST_OLD(DEBIAN_PHP_API)
 PHP_SUBST_OLD(EXTENSION_DIR)
 PHP_SUBST_OLD(EXTRA_LDFLAGS)
 PHP_SUBST_OLD(EXTRA_LDFLAGS_PROGRAM)
Index: php5-5.2.2/scripts/php-config.in
===================================================================
--- php5-5.2.2.orig/scripts/php-config.in	2007-05-04 17:08:28.000000000 +0200
+++ php5-5.2.2/scripts/php-config.in	2007-05-04 17:42:39.000000000 +0200
@@ -13,6 +13,7 @@
 program_suffix="@program_suffix@"
 exe_extension="@EXEEXT@"
 php_binary="@bindir@/${program_prefix}php${program_suffix}${exe_extension}"
+phpapi="@DEBIAN_PHP_API@"
 
 case "$1" in
 --prefix)
@@ -29,12 +30,14 @@
 	echo $includedir;;
 --php-binary)
 	echo $php_binary;;
+--phpapi)
+	echo $phpapi;;
 --version)
 	echo $version;;
 --vernum)
 	echo $version_id;;
 *)
-	echo "Usage: $0 [--prefix|--includes|--ldflags|--libs|--extension-dir|--include-dir|--php-binary|--version|--vernum]"
+	echo "Usage: $0 [--prefix|--includes|--ldflags|--libs|--extension-dir|--include-dir|--php-binary|--phpapi|--version|--vernum]"
 	exit 1;;
 esac
 
