Index: pkg-php/ext/sqlite/config.m4
===================================================================
--- pkg-php.orig/ext/sqlite/config.m4	2009-08-25 15:43:41.000000000 +0200
+++ pkg-php/ext/sqlite/config.m4	2009-08-25 15:44:03.000000000 +0200
@@ -84,7 +84,7 @@
     ])
     SQLITE_MODULE_TYPE=external
     PHP_SQLITE_CFLAGS=$pdo_inc_path
-    sqlite_extra_sources="libsqlite/src/encode.c"
+    sqlite_extra_sources=""
   else
     # use bundled library
     PHP_PROG_LEMON
