--- php-5.0.4/ext/standard/incomplete_class.c.bug34435	2004-11-25 20:28:37.000000000 +0000
+++ php-5.0.4/ext/standard/incomplete_class.c	2005-09-09 13:00:39.000000000 +0100
@@ -122,7 +122,7 @@
 
 /* {{{ php_lookup_class_name
  */
-char *php_lookup_class_name(zval *object, size_t *nlen)
+char *php_lookup_class_name(zval *object, zend_uint *nlen)
 {
 	zval **val;
 	char *retval = NULL;
@@ -144,7 +144,7 @@
 
 /* {{{ php_store_class_name
  */
-void php_store_class_name(zval *object, const char *name, size_t len)
+void php_store_class_name(zval *object, const char *name, zend_uint len)
 {
 	zval *val;
 	TSRMLS_FETCH();
--- php-5.0.4/ext/standard/php_incomplete_class.h.bug34435	2005-06-29 10:29:08.000000000 +0100
+++ php-5.0.4/ext/standard/php_incomplete_class.h	2005-09-09 13:00:31.000000000 +0100
@@ -42,7 +42,7 @@
 
 #define PHP_CLASS_ATTRIBUTES											\
 	char *class_name;													\
-	size_t name_len;													\
+	zend_uint name_len;													\
 	zend_bool free_class_name = 0;										\
 	zend_bool incomplete_class = 0
 
@@ -55,8 +55,8 @@
 	
 zend_class_entry *php_create_incomplete_class(TSRMLS_D);
 
-char *php_lookup_class_name(zval *object, size_t *nlen);
-void  php_store_class_name(zval *object, const char *name, size_t len);
+char *php_lookup_class_name(zval *object, zend_uint *nlen);
+void  php_store_class_name(zval *object, const char *name, zend_uint len);
 
 #ifdef __cplusplus
 };
