--- php5-5.1.1.0/ext/standard/datetime.c.orig	2005-08-04 00:07:57.000000000 +1000
+++ php5-5.1.1.0/ext/standard/datetime.c	2006-01-11 18:43:28.000000000 +1100
@@ -20,11 +20,11 @@
 
 /* $Id: datetime.c,v 1.134.2.2 2006/01/01 12:50:14 sniper Exp $ */
 
+#include "php.h"
 #if HAVE_STRPTIME
 #define _XOPEN_SOURCE
 #endif
 
-#include "php.h"
 #include "zend_operators.h"
 #include "datetime.h"
 #include "php_globals.h"
