--- php5-5.1.2/ext/reflection/config.m4.orig	2005-12-03 12:12:45.000000000 +1100
+++ php5-5.1.2/ext/reflection/config.m4	2006-01-16 16:04:28.000000000 +1100
@@ -2,7 +2,7 @@
 dnl config.m4 for extension reflection
 
 PHP_ARG_ENABLE(reflection, whether to enable reflection support,
-[  --disable-reflection    Disable reflection support], yes, no)
+[  --disable-reflection    Disable reflection support], yes)
 
 if test "$PHP_REFLECTION" != "no"; then
   if test "$ext_shared" = "yes"; then
