--- php5.old/ext/mbstring/config.m4	2006-10-27 15:58:10.000000000 +0200
+++ php5.new/ext/mbstring/config.m4	2006-10-27 15:59:09.000000000 +0200
@@ -293,7 +293,7 @@
   dnl libmbfl is required
   PHP_MBSTRING_SETUP_LIBMBFL
   PHP_MBSTRING_EXTENSION
-  PHP_INSTALL_HEADERS([ext/mbstring], [libmbfl/ libmbfl/mbfl])
+  PHP_INSTALL_HEADERS([ext/mbstring], [libmbfl/ libmbfl/mbfl/])
 fi
 
 # vim600: sts=2 sw=2 et
