--- php.orig/php.ini-development
+++ php/php.ini-development
@@ -333,6 +333,11 @@ allow_call_time_pass_reference = Off
 
 ; Safe Mode
 ; http://www.php.net/manual/en/ini.sect.safe-mode.php#ini.safe-mode
+; NOTE: this is considered a "broken" security measure.
+;       Applications relying on this feature will not recieve full
+;       support by the security team.  For more information please
+;       see /usr/share/doc/php5-common/README.Debian.security
+;
 safe_mode = Off
 
 ; By default, Safe Mode does a UID compare check when
@@ -374,6 +379,12 @@ safe_mode_protected_env_vars = LD_LIBRAR
 ; or per-virtualhost web server configuration file. This directive is
 ; *NOT* affected by whether Safe Mode is turned On or Off.
 ; http://www.php.net/manual/en/ini.sect.safe-mode.php#ini.open-basedir
+
+; NOTE: this is considered a "broken" security measure.
+;       Applications relying on this feature will not recieve full
+;       support by the security team.  For more information please
+;       see /usr/share/doc/php5-common/README.Debian.security
+;
 ;open_basedir =
 
 ; This directive allows you to disable certain functions for security reasons.
@@ -687,6 +698,11 @@ request_order = "GP"
 ; register_globals to be on;  Using form variables as globals can easily lead
 ; to possible security problems, if the code is not very well thought of.
 ; http://www.php.net/manual/en/ini.core.php#ini.register-globals
+
+; NOTE: applications relying on this feature will not recieve full
+;       support by the security team.  For more information please
+;       see /usr/share/doc/php5-common/README.Debian.security
+;
 register_globals = Off
 
 ; Determines whether the deprecated long $HTTP_*_VARS type predefined variables
