--- php.orig/configure.in
+++ php/configure.in
@@ -1,7 +1,7 @@
 ## $Id: configure.in 288372 2009-09-16 12:54:51Z iliaa $ -*- autoconf -*-
 dnl ## Process this file with autoconf to produce a configure script.
 
-divert(1)
+dnl divert(1)
 
 dnl ## Diversion 1 is the autoconf + automake setup phase. We also
 dnl ## set the PHP version, deal with platform-specific compile
@@ -23,16 +23,16 @@ dnl ## In diversion 5 we check which ext
 dnl ## All of these are normally in the extension directories.
 dnl ## Diversion 5 is the last one.  Here we generate files and clean up.
 
-dnl include Zend specific macro definitions first
-dnl -------------------------------------------------------------------------
-sinclude(Zend/acinclude.m4)
-
 dnl Basic autoconf + automake initialization, generation of config.nice.
 dnl -------------------------------------------------------------------------
 
 AC_PREREQ(2.13)
 AC_INIT(README.SVN-RULES)
 
+dnl include Zend specific macro definitions first
+dnl -------------------------------------------------------------------------
+sinclude(Zend/acinclude.m4)
+
 PHP_CONFIG_NICE(config.nice)
 
 PHP_CANONICAL_HOST_TARGET
@@ -263,7 +263,7 @@ sinclude(TSRM/threads.m4)
 sinclude(TSRM/tsrm.m4)
 
 
-divert(2)
+dnl divert(2)
 
 dnl ## Diversion 2 is where we set PHP-specific options and come up
 dnl ## with reasonable default values for them. We check for pthreads here
@@ -302,7 +302,7 @@ if test "$enable_maintainer_zts" = "yes"
   PTHREADS_FLAGS
 fi
 
-divert(3)
+dnl divert(3)
 
 dnl ## In diversion 3 we check for compile-time options to the PHP
 dnl ## core and how to deal with different system dependencies.
@@ -661,7 +661,7 @@ if test "x$php_crypt_r" = "x1"; then
   PHP_CRYPT_R_STYLE
 fi
 
-divert(4)
+dnl divert(4)
 
 dnl ## In diversion 4 we check user-configurable general settings.
 
@@ -902,7 +902,7 @@ else
   AC_MSG_RESULT([using system default])
 fi
 
-divert(5)
+dnl divert(5)
 
 dnl ## In diversion 5 we check which extensions should be compiled.
 dnl ## All of these are normally in the extension directories.
--- php.orig/ext/standard/config.m4
+++ php/ext/standard/config.m4
@@ -1,6 +1,6 @@
 dnl $Id: config.m4 286890 2009-08-06 14:07:16Z scottmac $ -*- autoconf -*-
 
-divert(3)dnl
+dnl divert(3)dnl
 
 dnl
 dnl Check if flush should be called explicitly after buffered io
@@ -205,7 +205,7 @@ dnl
 AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass isinf isnan)
 AC_FUNC_FNMATCH	
 
-divert(5)dnl
+dnl divert(5)dnl
 
 dnl
 dnl Check for regex library type
--- php.orig/scripts/phpize.m4
+++ php/scripts/phpize.m4
@@ -1,6 +1,6 @@
 dnl This file becomes configure.in for self-contained extensions.
 
-divert(1)
+dnl divert(1)
 
 AC_PREREQ(2.13)
 AC_INIT(config.m4)
