--- a/ext/standard/config.m4
+++ b/ext/standard/config.m4
@@ -235,7 +235,9 @@ main() {
 dnl
 dnl If one of them is missing, use our own implementation, portable code is then possible
 dnl
-if test "$ac_cv_crypt_blowfish" = "no" || test "$ac_cv_crypt_des" = "no" || test "$ac_cv_crypt_ext_des" = "no" || test "x$php_crypt_r" = "x0"; then
+dnl Don't use PHP own implementation now.
+dnl if test "$ac_cv_crypt_blowfish" = "no" || test "$ac_cv_crypt_des" = "no" || test "$ac_cv_crypt_ext_des" = "no" || test "x$php_crypt_r" = "x0"; then
+if false; then
 
   dnl
   dnl Check for __alignof__ support in the compiler
