--- a/ext/enchant/enchant.c	2010/10/16 17:30:27	304446
+++ b/ext/enchant/enchant.c	2010/10/16 17:52:01	304447
@@ -433,6 +433,8 @@
 	if (!value_len) {
 		RETURN_FALSE;
 	}
+	
+	PHP_ENCHANT_GET_BROKER;
 
 	switch (dict_type) {
 		case PHP_ENCHANT_MYSPELL:
@@ -466,6 +468,8 @@
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rl", &broker, &dict_type) == FAILURE) {
 		RETURN_FALSE;
 	}
+	
+	PHP_ENCHANT_GET_BROKER;
 
 	switch (dict_type) {
 		case PHP_ENCHANT_MYSPELL:
