--- php.orig/php.ini-dist
+++ php/php.ini-dist
@@ -510,7 +510,8 @@ user_dir =
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
 ; disabled on them.
-enable_dl = On
+; NOTE: this is a potential security hole and is disabled by default in debian
+enable_dl = Off
 
 ; cgi.force_redirect is necessary to provide security running PHP as a CGI under
 ; most web servers.  Left undefined, PHP turns this on by default.  You can
--- php.orig/php.ini-recommended
+++ php/php.ini-recommended
@@ -544,7 +544,9 @@ user_dir =
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
 ; disabled on them.
-enable_dl = On
+; NOTE: this is a potential security hole and is disabled by default in debian
+enable_dl = Off
+
 
 ; cgi.force_redirect is necessary to provide security running PHP as a CGI under
 ; most web servers.  Left undefined, PHP turns this on by default.  You can
