Index: php5-5.2.6/php.ini-dist
===================================================================
--- php5-5.2.6.orig/php.ini-dist
+++ php5-5.2.6/php.ini-dist
@@ -505,7 +505,8 @@ user_dir =
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
 ; disabled on them.
-enable_dl = On
+; NOTE: this is a potential security hole and is disabled by default in debian
+enable_dl = Off
 
 ; cgi.force_redirect is necessary to provide security running PHP as a CGI under
 ; most web servers.  Left undefined, PHP turns this on by default.  You can
Index: php5-5.2.6/php.ini-recommended
===================================================================
--- php5-5.2.6.orig/php.ini-recommended
+++ php5-5.2.6/php.ini-recommended
@@ -538,7 +538,9 @@ user_dir =
 ; Whether or not to enable the dl() function.  The dl() function does NOT work
 ; properly in multithreaded servers, such as IIS or Zeus, and is automatically
 ; disabled on them.
-enable_dl = On
+; NOTE: this is a potential security hole and is disabled by default in debian
+enable_dl = Off
+
 
 ; cgi.force_redirect is necessary to provide security running PHP as a CGI under
 ; most web servers.  Left undefined, PHP turns this on by default.  You can
