--- php5.orig/sapi/apache2filter/sapi_apache2.c
+++ php5/sapi/apache2filter/sapi_apache2.c
@@ -394,33 +394,24 @@ static int php_input_filter(ap_filter_t
 	return APR_SUCCESS;
 }
 
-static void php_apache_request_ctor(ap_filter_t *f, php_struct *ctx TSRMLS_DC)
+static int php_apache_request_ctor(ap_filter_t *f, php_struct *ctx TSRMLS_DC)
 {
-	char *content_type;
 	char *content_length;
 	const char *auth;
 
 	PG(during_request_startup) = 0;
 	SG(sapi_headers).http_response_code = !f->r->status ? HTTP_OK : f->r->status;
 	SG(request_info).content_type = apr_table_get(f->r->headers_in, "Content-Type");
-#undef safe_strdup
-#define safe_strdup(x) ((x)?strdup((x)):NULL)	
-	SG(request_info).query_string = safe_strdup(f->r->args);
+	SG(request_info).query_string = apr_pstrdup(f->r->pool, f->r->args);
 	SG(request_info).request_method = f->r->method;
 	SG(request_info).proto_num = f->r->proto_num;
-	SG(request_info).request_uri = safe_strdup(f->r->uri);
-	SG(request_info).path_translated = safe_strdup(f->r->filename);
+	SG(request_info).request_uri = apr_pstrdup(f->r->pool, f->r->uri);
+	SG(request_info).path_translated = apr_pstrdup(f->r->pool, f->r->filename);
 	f->r->no_local_copy = 1;
-	content_type = sapi_get_default_content_type(TSRMLS_C);
-	f->r->content_type = apr_pstrdup(f->r->pool, content_type);
-	SG(request_info).post_data = ctx->post_data;
-	SG(request_info).post_data_length = ctx->post_len;
-
-	efree(content_type);
 
 	content_length = (char *) apr_table_get(f->r->headers_in, "Content-Length");
 	SG(request_info).content_length = (content_length ? atol(content_length) : 0);
-	
+
 	apr_table_unset(f->r->headers_out, "Content-Length");
 	apr_table_unset(f->r->headers_out, "Last-Modified");
 	apr_table_unset(f->r->headers_out, "Expires");
@@ -435,7 +426,7 @@ static void php_apache_request_ctor(ap_f
 
 	ctx->r->user = apr_pstrdup(ctx->r->pool, SG(request_info).auth_user);
 
-	php_request_startup(TSRMLS_C);
+	return php_request_startup(TSRMLS_C);
 }
 
 static void php_apache_request_dtor(ap_filter_t *f TSRMLS_DC)
