--- a/Zend/zend_language_scanner.c
+++ b/Zend/zend_language_scanner.c
@@ -228,6 +228,12 @@ ZEND_API void zend_restore_lexical_state
 	SCNG(script_encoding) = lex_state->script_encoding;
 	SCNG(internal_encoding) = lex_state->internal_encoding;
 #endif /* ZEND_MULTIBYTE */
+
+	if (CG(heredoc)) {
+		efree(CG(heredoc));
+		CG(heredoc) = NULL;
+		CG(heredoc_len) = 0;
+	}
 }
 
 ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle TSRMLS_DC)
--- a/Zend/zend_language_scanner.l
+++ b/Zend/zend_language_scanner.l
@@ -227,6 +227,12 @@ ZEND_API void zend_restore_lexical_state
 	SCNG(script_encoding) = lex_state->script_encoding;
 	SCNG(internal_encoding) = lex_state->internal_encoding;
 #endif /* ZEND_MULTIBYTE */
+
+	if (CG(heredoc)) {
+		efree(CG(heredoc));
+		CG(heredoc) = NULL;
+		CG(heredoc_len) = 0;
+	}
 }
 
 ZEND_API void zend_destroy_file_handle(zend_file_handle *file_handle TSRMLS_DC)
--- /dev/null
+++ b/Zend/tests/bug53629.phpt
@@ -0,0 +1,31 @@
+--TEST--
+Bug #53629 (memory leak inside highlight_string())
+--FILE--
+<?php
+
+$str = '
+<?php
+class foo {
+    public $bar = <<<EOT
+bar
+    EOT;
+}
+?>
+';
+$str2 = '
+<?php
+var_dump(array(<<<EOD
+foobar!
+EOD
+));
+?>
+';
+
+highlight_string($str, true);
+highlight_string($str2, true);
+
+
+echo "Done\n";
+?>
+--EXPECT--
+Done
