Index: pkg-php/ext/soap/tests/server009.phpt
===================================================================
--- pkg-php.orig/ext/soap/tests/server009.phpt	2009-08-25 15:43:40.000000000 +0200
+++ pkg-php/ext/soap/tests/server009.phpt	2009-08-25 15:44:03.000000000 +0200
@@ -7,6 +7,10 @@
 		die('skip this test needs session extension');
 	}
 ?>
+--INI--
+session.auto_start=1
+session.save_handler=files
+session.save_path=temp_session_store
 --FILE--
 <?php
 class foo {
Index: pkg-php/ext/standard/tests/general_functions/phpinfo.phpt
===================================================================
--- pkg-php.orig/ext/standard/tests/general_functions/phpinfo.phpt	2009-08-25 15:44:03.000000000 +0200
+++ pkg-php/ext/standard/tests/general_functions/phpinfo.phpt	2009-08-25 15:44:03.000000000 +0200
@@ -1,5 +1,7 @@
 --TEST--
 phpinfo()
+--SKIPIF--
+<?php die("SKIP phpinfo - test suite's handling of "%s" is incompatible with this test"); ?>
 --FILE--
 <?php
 var_dump(phpinfo());
