Index: php5-5.2.5/tests/basic/bug20539.phpt
===================================================================
--- php5-5.2.5.orig/tests/basic/bug20539.phpt	2005-07-08 09:48:57.000000000 +0200
+++ php5-5.2.5/tests/basic/bug20539.phpt	2007-11-13 23:45:03.000000000 +0100
@@ -3,6 +3,7 @@
 --INI--
 session.auto_start=1
 session.save_handler=files
+session.save_path=temp_session_store
 --FILE--
 <?php
 	print "good :)\n";
Index: php5-5.2.5/ext/soap/tests/server009.phpt
===================================================================
--- php5-5.2.5.orig/ext/soap/tests/server009.phpt	2007-05-07 20:45:01.000000000 +0200
+++ php5-5.2.5/ext/soap/tests/server009.phpt	2007-11-13 23:45:03.000000000 +0100
@@ -7,6 +7,10 @@
 		die('skip this test needs session extension');
 	}
 ?>
+--INI--
+session.auto_start=1
+session.save_handler=files
+session.save_path=temp_session_store
 --FILE--
 <?php
 class foo {
Index: php5-5.2.5/ext/standard/tests/general_functions/phpinfo.phpt
===================================================================
--- php5-5.2.5.orig/ext/standard/tests/general_functions/phpinfo.phpt	2007-11-13 23:36:17.000000000 +0100
+++ php5-5.2.5/ext/standard/tests/general_functions/phpinfo.phpt	2007-11-13 23:45:03.000000000 +0100
@@ -1,5 +1,7 @@
 --TEST--
 phpinfo()
+--SKIPIF--
+<?php die("SKIP phpinfo - test suite's handling of "%s" is incompatible with this test"); ?>
 --FILE--
 <?php
 var_dump(phpinfo());
@@ -24,6 +26,7 @@
 Virtual Directory Support => %s
 Configuration File (php.ini) Path => %s
 Loaded Configuration File => %s
+Scan this dir for additional .ini files => %s
 PHP API => %d
 PHP Extension => %d
 Zend Extension => %d
Index: php5-5.2.5/ext/standard/tests/strings/moneyformat.phpt
===================================================================
--- php5-5.2.5.orig/ext/standard/tests/strings/moneyformat.phpt	2007-06-06 19:46:17.000000000 +0200
+++ php5-5.2.5/ext/standard/tests/strings/moneyformat.phpt	2007-11-13 23:45:03.000000000 +0100
@@ -5,6 +5,8 @@
 	if (!function_exists('money_format') || !function_exists('setlocale')) {
 		die("SKIP money_format - not supported\n");
 	}
+	if (!setlocale(LC_MONETARY, 'en_US'))
+		die("SKIP money_format - en_US locale not available\n");
 ?>
 --FILE--
 <?php
