--- a/tests/lang/bug30638.phpt
+++ b/tests/lang/bug30638.phpt
@@ -2,7 +2,7 @@
 Bug #30638 (localeconv returns wrong LC_NUMERIC settings) (ok to fail on MacOS X)
 --SKIPIF--
 <?php  # try to activate a german locale
-if (setlocale(LC_NUMERIC, "de_DE", "de", "german", "ge") === FALSE) {
+if (setlocale(LC_NUMERIC, "de_DE.UTF-8", "de_DE", "de", "german", "ge", "de_DE.ISO-8859-1") === FALSE) {
 	print "skip setlocale() failed";
 } elseif (strtolower(php_uname('s')) == 'darwin') {
     print "skip ok to fail on MacOS X";
@@ -11,7 +11,7 @@ if (setlocale(LC_NUMERIC, "de_DE", "de",
 --FILE--
 <?php 
 # activate the german locale
-setlocale(LC_NUMERIC, "de_DE", "de", "german", "ge");
+setlocale(LC_NUMERIC, "de_DE.UTF-8", "de_DE", "de", "german", "ge", "de_DE.ISO-8859-1");
 
 $lc = localeconv();
 printf("decimal_point: %s\n", $lc['decimal_point']);
--- a/tests/lang/034.phpt
+++ b/tests/lang/034.phpt
@@ -4,14 +4,14 @@ Bug #12647 (Locale settings affecting fl
 precision=14
 --SKIPIF--
 <?php  # try to activate a german locale
-if (setlocale(LC_NUMERIC, "de_DE", "de", "german", "ge", "de_DE.ISO8859-1") === FALSE) {
+if (setlocale(LC_NUMERIC, "de_DE.UTF-8", "de_DE", "de", "german", "ge", "de_DE.ISO-8859-1") === FALSE) {
 	print "skip";
 }
 ?>
 --FILE--
 <?php 
 # activate the german locale
-setlocale(LC_NUMERIC, "de_DE", "de", "german", "ge", "de_DE.ISO8859-1");
+setlocale(LC_NUMERIC, "de_DE.UTF-8", "de_DE", "de", "german", "ge", "de_DE.ISO-8859-1");
 
 echo (float)"3.14", "\n";
 ?>
