Index: php/ext/pdo_dblib/dblib_driver.c
===================================================================
--- php.orig/ext/pdo_dblib/dblib_driver.c
+++ php/ext/pdo_dblib/dblib_driver.c
@@ -230,9 +230,11 @@ static int pdo_dblib_handle_factory(pdo_
 		goto cleanup;
 	}
 
+#if PHP_DBLIB_IS_MSSQL
 	if (DBSETOPT(H->link, DBTEXTLIMIT, "2147483647") == FAIL) {
 		goto cleanup;
 	}
+#endif
 
 	if (vars[3].optval && FAIL == dbuse(H->link, vars[3].optval)) {
 		goto cleanup;
