Index: php/pear/install-pear-nozlib.phar
===================================================================
--- php.orig/pear/install-pear-nozlib.phar
+++ php/pear/install-pear-nozlib.phar
@@ -35998,7 +35998,7 @@ if (getenv('PHP_PEAR_SYSCONF_DIR')) {
 } elseif (getenv('SystemRoot')) {
     define('PEAR_CONFIG_SYSCONFDIR', getenv('SystemRoot'));
 } else {
-    define('PEAR_CONFIG_SYSCONFDIR', PHP_SYSCONFDIR);
+    define('PEAR_CONFIG_SYSCONFDIR', PHP_SYSCONFDIR . DIRECTORY_SEPARATOR . 'pear');
 }
 
 // Default for master_server
