Index: php/ext/sockets/sockets.c
===================================================================
--- php.orig/ext/sockets/sockets.c
+++ php/ext/sockets/sockets.c
@@ -229,7 +229,7 @@ static int php_open_listen_sock(php_sock
 	sock->type = PF_INET;
 
 	if (bind(sock->bsd_socket, (struct sockaddr *)&la, sizeof(la)) < 0) {
-		PHP_SOCKET_ERROR(sock, "unable to bind to given adress", errno);
+		PHP_SOCKET_ERROR(sock, "unable to bind to given address", errno);
 		close(sock->bsd_socket);
 		efree(sock);
 		return 0;
