Description: Fix memory leak in cyclical garbage collector
Origin: http://svn.php.net/viewvc?view=revision&revision=298213
Last-Update: 2010-05-01

Index: php/Zend/zend_execute.c
===================================================================
--- php.orig/Zend/zend_execute.c
+++ php/Zend/zend_execute.c
@@ -708,6 +708,7 @@ static inline zval* zend_assign_to_varia
 				return variable_ptr;
 			}
 		} else { /* we need to split */
+			GC_ZVAL_CHECK_POSSIBLE_ROOT(*variable_ptr_ptr);
 			if (!is_tmp_var) {
 				if (PZVAL_IS_REF(value) && Z_REFCOUNT_P(value) > 0) {
 					ALLOC_ZVAL(variable_ptr);
