Description: Fix mysqli_ssl_set
Origin: http://svn.php.net/viewvc?view=revision&revision=297688
Last-Update: 2010-05-01

Index: php/ext/mysqli/mysqli_api.c
===================================================================
--- php.orig/ext/mysqli/mysqli_api.c
+++ php/ext/mysqli/mysqli_api.c
@@ -2134,7 +2134,7 @@ PHP_FUNCTION(mysqli_ssl_set)
 	if (zend_parse_method_parameters(ZEND_NUM_ARGS() TSRMLS_CC, getThis(), "Osssss", &mysql_link, mysqli_link_class_entry, &ssl_parm[0], &ssl_parm_len[0], &ssl_parm[1], &ssl_parm_len[1], &ssl_parm[2], &ssl_parm_len[2], &ssl_parm[3], &ssl_parm_len[3], &ssl_parm[4], &ssl_parm_len[4])   == FAILURE) {
 		return;
 	}
-	MYSQLI_FETCH_RESOURCE(mysql, MY_MYSQL *, &mysql_link, "mysqli_link", MYSQLI_STATUS_VALID);
+	MYSQLI_FETCH_RESOURCE(mysql, MY_MYSQL *, &mysql_link, "mysqli_link", MYSQLI_STATUS_INITIALIZED);
 
 	for (i = 0; i < 5; i++) {
 		if (!ssl_parm_len[i]) {
