Description: Fix memory leak in the soap extension
Origin: http://svn.php.net/viewvc?view=revision&revision=298433
Last-Update: 2010-05-02

Index: php/ext/soap/soap.c
===================================================================
--- php.orig/ext/soap/soap.c
+++ php/ext/soap/soap.c
@@ -1699,6 +1699,7 @@ PHP_METHOD(SoapServer, handle)
 				doc_request = soap_xmlParseMemory(post_data, post_data_length);
 			}
 		} else {
+			zval_ptr_dtor(&retval);
 			return;
 		}
 	} else {
